src/WebJobs.Extensions.DurableTask/Listener/TaskEntityShim.cs

```diff
@@ -290,9 +331,32 @@ async Task ExecuteOperationsInBatch()
                     for (this.BatchPosition = 0; this.BatchPosition < this.operationBatch.Count; this.BatchPosition++)
                     {
                         // first, check if we should stop here and not execute the rest of the batch
-                        if (onHostStopping.IsCancellationRequested // host is shutting down
-                            || this.TimeoutTask.IsCompleted // we have timed out
-                            || stopwatch.Elapsed > TimeSpan.FromMinutes(1)) // we have spent significant time in this batch
+                        bool InterruptBatchProcessing()
+                        {
+                            if (onHostStopping.IsCancellationRequested)
+                            {
+                                // host is shutting down
+                                return true;
+                            }
+                            else if (this.TimeoutTask.IsCompleted)
+                            {
+                                // we timed out
+                                this.AddTraceFlag(EntityTraceFlags.TimedOut);
+                                return true;
+                            }
+                            else if (stopwatch.Elapsed > TimeSpan.FromMinutes(3))
+                            {
+                                // we have spent a significant time on this batch already
+                                this.AddTraceFlag(EntityTraceFlags.SignificantTimeElapsed);
+                                return true;
+                            }
+                            else
+                            {
+                                return false;
+                            }
+                        }
```

Comment:
stylistic nit: any reason for this is a local function? I agree it's helpful to have it abstracted out as a named method, but I think that this is _just_ long enough to be refactored into a class-level method.

Reply:
I suppose it is a matter of preference. I like local functions because code that belongs together logically stays in close proximity in the source file which helps me to read it. For example, I don't have to define and pass "stopwatch" or "onHostStopping" as arguments, and when you read the function you can directly see where those variables come from.

Reply:
Fair enough, it is indeed just a matter of preference. I won't block on this.